vm/boolean_test.go

```diff
@@ -41,6 +41,14 @@ func TestEvalInfixBooleanExpression(t *testing.T) {
 		{"(1 < 2) == false", false},
 		{"(1 > 2) == true", false},
 		{"(1 > 2) == false", true},
+		{"true && true", true},
+		{"false && true", false},
+		{"true && false", false},
+		{"false && false", false},
+		{"true || true", true},
+		{"false || true", true},
+		{"true || false", true},
+		{"false || false", false},
```

Comment:
@janczer Can you add more complex condition to test new precedence setting? Like:

```ruby
100 > 10 && true == false # false
```